include/dem/localize_contacts.h

```diff
@@ -76,4 +74,29 @@ localize_contacts(
     dim>::particle_floating_mesh_candidates
     &particle_floating_mesh_contact_candidates);
 
+/**
+ * Manages removing repetitions and adding new contact pairs to the contact
+ * containers when particles are exchanged between processors. If the contact
+ * pair (in adjacent particles containers) does not exist in the output of the
+ * new (current step) broad search, it is removed from the contact pair
+ * (adjacent containers), since it means that the contact is being handled by
+ * another processor. If the pair exists in the output of the new broad search,
+ * it is removed from the output of the broad search, as the contact is already
+ * being processed. This process is performed for many types of particle-object
+ * pairs : local-local particle-particle pairs, local-ghost particle-particle
+ * pairs and particle-wall pairs, particle-floating wall pairs and
+ * particle-floating mesh contacts pairs. Note : contact_type is important for
+ * local or ghost particle-particle contact only
+ *
+ * @param pairs_in_contact adjacent particle-object pairs from fine search at last time step
+ * @param contact_candidates particle-object contact pairs from board search at current time step
+ * @param contact_type label of contact type to apply proper manipulation of contact removal in containers
+ */
+
+template <int dim, typename pairs_structure, typename candidates_structure>
+void
+update_fine_search_candidates(pairs_structure &     pairs_in_contact,
+                              candidates_structure &contact_candidates,
+                              const ContactType     contact_type);
```

Comment:
I would make the contact_type a template parameter, this way you can use a if constexpr instead... We always know what type of contact type we want to flush before calling it anyway.